### PR TITLE
[spmd] Add a few more loss ops to the reduction op list

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99964
* #99901
* __->__ #99900
* #99899
* #99898
* #99374
* #99489
* #99436
* #99897

This PR adds a few more loss ops to the reduction op list

### CODE REVIEW

File: torch/distributed/_spmd/data_parallel.py [hunk 1]
```diff
@@ -155,6 +155,19 @@ def __init__(self, batch_dim: int = 0) -> None:
...
         self.batch_dim_size = -1
 
+        # reduction ops that is used to detect whether there's a reduction over batch
+        # dimension operation, if there is, we mark the output as sharded instead of
+        # partial placement
+        self.reduction_ops = [
+            aten.binary_cross_entropy.default,
+            aten.binary_cross_entropy_with_logits.default,
+            aten.mean.default,
+            aten.mse_loss.default,
```
Comment (mrshenli) [inline torch/distributed/_spmd/data_parallel.py:165]: do we need to test the arguments? If `reduction=None`, it seems there won't be a reduction?

https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html

**Review by mrshenli — APPROVED** [review]
stamp to unblock

File: torch/distributed/_spmd/data_parallel.py (hunk 1 above)
Comment (wanchaol) [reply torch/distributed/_spmd/data_parallel.py:165]: I see, thinking about it more, I think it should be ok for us, we use `reduction_ops` and whether output_tensor becomes a scalar tensor to jointly determine if there's a reduction on batch dim, so if `reduction=None` it would not be mark as reduction over batch dim
